macOS/DuckDuckGo/MainWindow/MainViewController.swift

```diff
@@ -485,6 +501,63 @@ final class MainViewController: NSViewController {
         NSApp.mainMenuTyped.stopMenuItem.isEnabled = selectedTabViewModel.isLoading
     }
 
+    // MARK: - Set As Default and Add To Dock Prompts configuration
+
+    var isBannerViewVisible: Bool {
+        mainView.bannerHeightConstraint.constant != 0
+    }
+
+    private func subscribeToSetAsDefaultAndAddToDockPromptsNotifications() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(tryToShowSetAsDefaultAndAddtoDockIfNeeded),
+                                               name: .showPromptForSetAsDefaultBrowserAndAddToDock,
+                                               object: nil)
+
+        bannerDismissedCancellable = defaultBrowserAndDockPromptPresenting.bannerDismissedPublisher
+            .sink { [weak self] in
+                self?.hideBanner()
+            }
+    }
+
+    @objc private func tryToShowSetAsDefaultAndAddtoDockIfNeeded() {
+        defaultBrowserAndDockPromptPresenting.tryToShowPrompt(
+            popoverAnchorProvider: getSourceViewToShowSetAsDefaultAndAddToDockPopover,
+            bannerViewHandler: showMessageBanner
+        )
+    }
+
+    private func getSourceViewToShowSetAsDefaultAndAddToDockPopover() -> NSView? {
+        if self.isViewLoaded && self.view.window?.isKeyWindow == true {
+            if bookmarksBarVisibilityManager.isBookmarksBarVisible {
+                return self.bookmarksBarViewController.view
+            } else {
+                return self.navigationBarViewController.addressBarViewController?.view
+            }
+        }
+
+        return nil
```

Comment:
Replacing the `if` with `guard` could make this simpler by removing nested ifs.

```suggestion
        guard isViewLoaded && view.window?.isKeyWindow == true else {
            return nil
        }
        if bookmarksBarVisibilityManager.isBookmarksBarVisible {
            return bookmarksBarViewController.view
        } else {
            return navigationBarViewController.addressBarViewController?.view
        }
```